syntax = "proto3";
package user;

service UserService {
  // Регистрация нового пользователя
  rpc Register(RegisterRequest) returns (RegisterResponse);
  // Авторизация
  rpc Login(LoginRequest) returns (LoginResponse);
  // Получить профиль
  rpc GetProfile(GetProfileRequest) returns (UserProfile);
}

message RegisterRequest {
  string username = 1;
  string email = 2;
  string password = 3;
}
message RegisterResponse {
  string user_id = 1;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}
message LoginResponse {
  string token = 1;
}

message GetProfileRequest {
  string user_id = 1;
}
message UserProfile {
  string user_id = 1;
  string username = 2;
  string email = 3;
  string display_name = 4;
  string avatar_url = 5;
  int64 created_at = 6;
  int64 updated_at = 7;
}
